src/cpp/src/index_partition.cpp

```diff
@@ -99,6 +110,52 @@ void IndexPartition::remove(int64_t index) {
     ids_[index] = ids_[last_idx];
 
     num_vectors_--;
+
+    removeAttribute(index);
+}
+
+// https://github.com/apache/arrow/issues/44243
+// Arrow data is immutable. So you can't delete a row from existing Arrow data.
+// You need to create a new Arrow data that doesn't have the target row.
+void IndexPartition::removeAttribute(int64_t target_id) {
```

Comment:
Instead of this we should maintain a set of tombstones which mark if an entry is deleted.

So we maintain a vector<bool> for the partitions and if we delete an entry we mark the tombstone as 0. Then, during maintenance, we do compaction.

However, for your purposes you can ignore removals for attribute filtering to simplify your implementation. **So only consider adds, and searches for now and we wil deal with removes later**